pkgs/brew/xnvme.rb

```diff
@@ -0,0 +1,69 @@
+class Xnvme < Formula
+  desc "Cross-platform libraries and tools for efficient I/O and low-level control"
+  homepage "https://xnvme.io/"
+  url "https://github.com/OpenMPDK/xNVMe/releases/download/v0.7.2/xnvme-0.7.2.tar.gz"
+  sha256 "1cb849b537cfddc15d82b8f4622fe3f999b4c7c0542c55b8d09b485e016e942e"
+  license "BSD-3-Clause"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => [:build, :test]
+  depends_on "python@3.12" => :build
+
+  def install
+    # We do not have SPDK nor libvfn on macOS, thus disabling these
+    # The examples and tests are also a bit superflous, so disable those as well
+    system "meson", "setup", "build",
+           *std_meson_args,
+           "-Dwith-spdk=false",
+           "-Dwith-libvfn=disabled",
+           "-Dtests=false",
+           "-Dexamples=false"
+    system "meson", "compile", "-C", "build"
+    system "meson", "install", "-C", "build"
+  end
+
+  test do
+    # Verify cli usage using a "ramdisk" of 1GB
+    output = shell_output("#{bin}/xnvme library-info")
+    assert_match "XNVME_BE_RAMDISK_ENABLED", output
+
+    output = shell_output("#{bin}/xnvme info 1GB --be ramdisk")
+    assert_match "uri: '1GB'", output
+    assert_match "type: XNVME_GEO_CONVENTIONAL", output
+    assert_match "tbytes: 1073741824", output
+
+    # Verify library usage using a ramdisk of 1GB
+    (testpath/"test.c").write <<~EOS
+      #include <stdio.h>
+      #include <libxnvme.h>
+
+      int main(int argc, char **argv) {
+        struct xnvme_opts opts = xnvme_opts_default();
+        struct xnvme_dev *dev;
+
+        dev = xnvme_dev_open("1GB", &opts);
+        if (!dev) {
+          perror("xnvme_dev_open()");
+          return 1;
+        }
+
+        xnvme_dev_pr(dev, XNVME_PR_DEF);
+        xnvme_dev_close(dev);
+
+        return 0;
+      }
+    EOS
+
+    # Build the example using pkg-config for build-options
+    flags = shell_output("pkg-config xnvme --libs --cflags").strip
+    system ENV.cc, "test.c", "-o", "test", *flags.split
+
+    # Run it and check the output, this should produce the same output as the
+    # 'xnvme library-info' command, thus the output-assertion is the same
+    output = shell_output("./test info 1GB --be ramdisk")
```

Comment:
You are not using the arguments you give here

Reply:
I don't think I follow?

The content of ``output`` is asserted. E.g. checking that expected output is produced from running the command.

Reply:
The program you wrote, `test`,  doesn't take any arguments, but you provide `info 1GB --be ramdisk`

Reply:
Ohh, you are right, I have made a note on https://github.com/OpenMPDK/xNVMe/issues/389 to fix this with the next release. The "xnvme.rb" in this PR is the version that first went into ``brew``:

https://github.com/Homebrew/homebrew-core/blob/master/Formula/x/xnvme.rb

The intent is to just keep the file to show where it came from and future work to be done on the ``homebrew-core`` repository.